src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,202 @@
+        if (currentNode == null) {
+            return false;
+        }
+        assignPreviousAndNextNode();
+        unlink(currentNode);
+        size--;
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void iterateCurrentNode(int index) {
+        currentNode = head;
+        for (int i = 0; i < index; i++) {
+            currentNode = currentNode.next;
+        }
+    }
+
+    private void verifyIndex(int index) {
+        if (index < 0 || index > size - 1) {
+            throw new IndexOutOfBoundsException("The index passed must be equal "
+                    + "or greater than 0 and less than " + size);
+        }
+    }
+
+    private void addNodeInTheMiddle(T value) {
+        previousNode = currentNode.prev;
+        Node<T> newElement = new Node<>(value);
+        newElement.prev = previousNode;
+        newElement.next = currentNode;
+        if (currentNode == head || currentNode != null && currentNode.equals(head)) {
+            head = newElement;
+        }
+        if (previousNode != null) {
+            previousNode.next = newElement;
+        }
+        currentNode.prev = newElement;
+        size++;
+    }
+
+    private T removeFirstNode() {
+        final Node<T> removedElement = head;
+        head = head.next;
+        if (head != null) {
+            head.prev = null;
+        }
+        size--;
+        return removedElement.item;
+    }
+
+    private void assignPreviousAndNextNode() {
+        previousNode = currentNode.prev;
+        nextNode = currentNode.next;
+        if (nextNode != null) {
+            nextNode.prev = previousNode;
+        }
+        if (previousNode != null) {
+            previousNode.next = nextNode;
+        }
+    }
+
+    private void unlink(Node<T> node) {
+        node.prev = null;
+        node.next = null;
+        if (tail.equals(node)) {
+            Node<T> newNode = tail;
+            newNode.next = null;
+            newNode.prev = null;
+        }
+        if (head.equals(node)) {
+            Node<T> newNode = head;
+            newNode.next = null;
+            newNode.prev = null;
+        }
+    }
+
+    private void findInTheMiddle(T object) {
+        currentNode = head;
+        while (currentNode != null) {
+            if (currentNode.item == object || currentNode.item != null
+                    && currentNode.item.equals(object)) {
+                break;
+            }
+            currentNode = currentNode.next;
+        }
+    }
+
+    private boolean removeFromTheHead() {
+        head = head.next;
+        if (head != null) {
+            head.prev = null;
+        }
+        size--;
+        return true;
+    }
+
+    public static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(T item) {
```

Comment:
add all parameters to constructor 
read [common mistakes file](https://github.com/mate-academy/jv-linked-list/blob/master/checklist.md)